src/profiler/storage_profiler.h

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#ifndef MXNET_PROFILER_STORAGE_PROFILER_H_
+#define MXNET_PROFILER_STORAGE_PROFILER_H_
+
+#include <mxnet/storage.h>
+#include <string>
+#include <vector>
+#include "./profiler.h"
+
+namespace mxnet {
+namespace storage {
+
+#if MXNET_USE_PROFILER
```

Comment:
Note for future reference: we should consider removing this macro and always compile profiler if it doesn't introduce overhead when profiler is off.

One possibility of overhead even when profiler is off might comes from PROFILER_MESSAGE strings.

Reply:
ok